Rover/system.cpp

```diff
@@ -136,6 +136,11 @@ void Rover::init_ardupilot()
 
     rover.g2.sailboat.init();
 
+    // boat should loiter after completing a mission to avoid drifting off
+    if (is_boat()) {
+        AP_Param::set_default_by_name("MIS_DONE_BEHAVE", 1);
```

Comment:
```suggestion
        g2.mis_done_behave.set_default(MIS_DONE_BEHAVE_LOITER);
```
saves the string lookup and used the enum value to set.

Reply:
Ah right, because the parameter is visible to the vehicle class this works.  Seems like a good suggestion to me.

Reply:
Ok. I'll change it to use `set_default`. 
The  `Mis_Done_Behave` is private to `ModeAuto` class. Should I make it public?

Reply:
@shiv-tyagi, yes, making the enum public seems OK to me. txs

Reply:
Done!